common/src/main/java/org/apache/uniffle/common/netty/protocol/SendShuffleDataRequestV1.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.netty.protocol;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import io.netty.buffer.ByteBuf;
+
+import org.apache.uniffle.common.ShufflePartitionedBlock;
+import org.apache.uniffle.common.util.ByteBufUtils;
+
+public class SendShuffleDataRequestV1 extends RequestMessage {
```

Comment:
The earliest version was a modification of the original class and there will be a lot of intrusive code and it will not be easy to review and understand.

SendShuffleDataRequestV1 class will not decode useless information to ShuffleBlockInfo which is temporary.

Reply:
Will we have v2? If we don't have v2, maybe we should remove v1.

Reply:
`SendShuffleDataRequestV1` will be only used by Shuffle Server and `SendShuffleDataRequest` only be used by client.  It is compatible with currently running cluster and serves as a transition.

Next stage, i will add `SendShuffleDataRequestV2` which will be used by Shuffle Server and client. `SendShuffleDataRequestV2` will not contain useless information which is sent from client.

Reply:
The request of the client is different from the one of the server. It's a little weird.

Reply:
@jerqi Since @lwllvyb get way from work this week, I reply it instead of him.

In our production scenario, we cannot ensure the upper compute product use our newest version at the same time, even we cannot ensure we upgrade all of the rss client at the same time, most of time, we update some of them.

So the server have to handle old/new client request, as we all know, there are some memory/cpu waste about the old `SendShuffleDataRequest`, so the main motivation of this PR is to optimize this point, we have to take care of the old/new version client, so we invent `SendShuffleDataRequestV1` to handle the request from old client, and after this PR, we are ready to implement `SendShuffleDataRequestV2` to handle the new request from new client, this is a trade off between `clear code` and `consistency`.

I found there are some similar case in the existing code, e.g. ShuffleTaskManager#requireBuffer, CoordinatorGrpcService#fetchClientConfV2 and ApplicationManager#refreshAppId and so on.

Reply:
It's ok although it's better that we use server version to solve upgrade issue. Coordinator must be compatible with old version client. Shuffle server can be used shuffle server version to upgrade.

Reply:
Yeah, we'd like to keep client version newer as mush as possible, but it's hard to inspire all user replace our new release version client jar in their image.

If there are any suggestion about this PR, I can updated it by me.

Reply:
In Tencent, you can upgrade every client. So we chose that solution at that time.